plans/benchmarks/benchmarks.go

```diff
@@ -0,0 +1,164 @@
+		ImprovementDir: -1,
+	},
+		duration.Seconds())
+}
+
+// StartTimeBench does nothing but start up and report the time it took to start.
+// This relies on the testground daemon to inject the time when the plan is scheduled
+// into the runtime environment
+func StartTimeBench(runenv *runtime.RunEnv) error {
+	scheduledTime := runenv.TestStartTime
+	startupTime := time.Now()
+	emitTime(runenv, "Time to Start", startupTime.Sub(scheduledTime))
+	return nil
+}
+
+// NetworkInitBench starts and waits for the network to initialize
+// The metric it emits represents the time between plan start and when the network initialization
+// is completed.
+func NetworkInitBench(runenv *runtime.RunEnv) error {
+	startupTime := time.Now()
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+	watcher, writer := sync.MustWatcherWriter(ctx, runenv)
+	defer watcher.Close()
+	defer writer.Close()
+
+	if err := sync.WaitNetworkInitialized(ctx, runenv, watcher); err != nil {
+		return err
+	}
+
+	emitTime(runenv, "Time to Network", time.Now().Sub(startupTime))
+	return nil
+}
+
+func NetworkLinkShapeBench(runenv *runtime.RunEnv) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+	watcher, writer := sync.MustWatcherWriter(ctx, runenv)
+	defer watcher.Close()
+	defer writer.Close()
+
+	if err := sync.WaitNetworkInitialized(ctx, runenv, watcher); err != nil {
+		return err
+	}
+	// A state name unique to the container...
+	name, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+	doneState := sync.State("net configured " + name)
+
+	// A new network configuration
+	netConfig := sync.NetworkConfig{
+		Network: "default",
+		Default: sync.LinkShape{
+			Latency: 250 * time.Millisecond,
+		},
+		State: doneState,
+	}
+
+	beforeNetConfig := time.Now()
+	// Send configuration to the sidecar.
+	writer.Write(ctx, sync.NetworkSubtree(name), &netConfig)
+	// Wait for the signal that the network change is completed.
+	err = <-watcher.Barrier(ctx, doneState, 1)
+	if err != nil {
+		return err
+	}
+	emitTime(runenv, "Time to configure link shape", time.Now().Sub(beforeNetConfig))
+	return nil
+}
+
+func NetworkIpChangeBench(runenv *runtime.RunEnv) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 300*time.Second)
+	defer cancel()
+	watcher, writer := sync.MustWatcherWriter(ctx, runenv)
+	defer watcher.Close()
+	defer writer.Close()
+
+	if err := sync.WaitNetworkInitialized(ctx, runenv, watcher); err != nil {
+		return err
+	}
+	// A state name unique to the container...
+	name, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+	doneState := sync.State("net configured " + name)
+
+	// A new network configuration
+	netConfig := sync.NetworkConfig{
+		Network: "default",
+		IPv4:    &runenv.TestSubnet.IPNet,
+		State:   doneState,
+	}
+	// Change the IP address.
+	// Not checking if the IP address I'm changing to already exists, by the way
```

Comment:
I don't think we can do that, this is expected to fail, but it would be good to know how exactly it fails. In any case I wouldn't spend too much time on it now, as manual IP changing from testplans is not on the critical path for the short term.

Reply:
I *can* do this on local docker... but you're right on k8s... Taking it out!